dali/operators/generic/erase/erase.cu

```diff
@@ -0,0 +1,162 @@
+        kmgr_.Resize<EraseKernel>(1, 1);
+      }
+
+  bool SetupImpl(std::vector<OutputDesc> &output_desc,
+                 const workspace_t<GPUBackend> &ws) override {
+    const auto &input = ws.template InputRef<GPUBackend>(0);
+    auto layout = input.GetLayout();
+    auto type = input.type();
+    auto shape = input.shape();
+    AcquireArgs(ws, shape, layout);
+
+    auto in_view = view<const T, Dims>(input);
+    kmgr_.Initialize<EraseKernel>();
+    ctx_.gpu.stream = ws.stream();
+    kmgr_.Setup<EraseKernel>(0, ctx_, in_view, regions_, make_cspan(fill_values_));
+
+    output_desc.resize(1);
+    output_desc[0] = {shape, input.type()};
+    return true;
+  };
+
+  void RunImpl(workspace_t<GPUBackend> &ws) override {
+    auto input = view<const T, Dims>(ws.template InputRef<GPUBackend>(0));
+    auto output = view<T, Dims>(ws.template OutputRef<GPUBackend>(0));
+    kmgr_.Run<EraseKernel>(0, 0, ctx_, output, input, regions_, make_cspan(fill_values_));
+  }
+
+  void AcquireArgs(const ArgumentWorkspace &ws, TensorListShape<> in_shape,
+                   TensorLayout in_layout) {
+    fill_values_ = spec_.template GetRepeatedArgument<float>("fill_value");
+    DALI_ENFORCE(channel_dim >= 0 || fill_values_.size() <= 1,
+      "If a multi channel fill value is provided, the input layout must have a 'C' dimension");
+    regions_.resize(batch_size_);
+    auto args = detail::GetEraseArgs<T, Dims>(spec_, ws, in_shape, in_layout);
+    for (int i = 0; i < batch_size_; ++i) {
+      regions_[i].reserve(args[i].rois.size());
+      const auto &rois = args[i].rois;
+      for (const auto &roi : args[i].rois) {
+        regions_[i].push_back(detail::make_box(roi.anchor, roi.shape));
+      }
+    }
+  }
+
+ private:
+  OpSpec spec_;
+  int batch_size_ = 0;
+  std::vector<int> axes_;
+  std::vector<kernels::EraseArgs<T, Dims>> args_;
+  std::vector<std::vector<kernels::ibox<Dims>>> regions_;
+  SmallVector<T, 3> fill_values_;
+  kernels::KernelManager kmgr_;
+  kernels::KernelContext ctx_;
+};
+
+namespace detail {
+
+template <typename T, int Dims, int... ChDims>
+struct select_impl {};
+
+template <typename T, int Dims, int ChDim, int... ChDims>
+struct select_impl<T, Dims, ChDim, ChDims...> {
+  static std::unique_ptr<OpImplBase<GPUBackend>> make(int channel_dim, const OpSpec &spec) {
+    if (channel_dim == ChDim) {
+      return std::make_unique<EraseImplGpu<T, Dims, ChDim>>(spec);
+    } else {
+      return select_impl<T, Dims, ChDims...>::make(channel_dim, spec);
+    }
+  }
+};
+
+template <typename T, int Dims, int ChDim>
+struct select_impl<T, Dims, ChDim> {
+  static std::unique_ptr<OpImplBase<GPUBackend>> make(int channel_dim, const OpSpec &spec) {
+    if (channel_dim == ChDim) {
+      return std::make_unique<EraseImplGpu<T, Dims, ChDim>>(spec);
+    } else {
+      DALI_FAIL("Unsopported layout. Only channel-first and channel-last layouts are supported.");
+    }
+  }
+};
```

Comment:
Maybe add a comment that it's used to select one of the allowed ChannelDim positions for those of a fainter heart.